src/main/java/rx/Single.java

```diff
@@ -211,6 +211,21 @@ protected Single(final Observable.OnSubscribe<T> f) {
      */
 
     /**
+     * Casts the success value of the current Single into the target type or signals a
+     * ClassCastException if not compatible.
+     * <dl>
+     * <dt><b>Scheduler:</b></dt>
+     * <dd>{@code cast} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param <R> the target type
+     * @param klass the type token to use for casting the success result from the current Single
+     * @return the new Single instance
+     */
+    public final <R> Single<R> cast(final Class<R> klass) {
+        return lift(new OperatorCast<T, R>(klass));
```

Comment:
@akarnokd I will update the PR with a simplified implementation.  Thanks for the suggestion.